### PR TITLE
interfaces/testiface: added TestHotplugInterface

### PR DESCRIPTION
**Pull request by stolowski**
Extend TestInterface with TestHotplugInterface. It's used in tests of the upcoming PR(s).

Note: it depends on #5841, it will fail until it lands.

### CODE REVIEW

File: interfaces/ifacetest/testiface.go [hunk 1]
```diff
@@ -104,6 +105,16 @@ type TestInterface struct {
 	SystemdPermanentSlotCallback func(spec *systemd.Specification, slot *snap.SlotInfo) error
 }
 
+// TestHotplugInterface is a interface for various kind of tests needing hotplug-aware interface.
```
Comment (bboozzoo) [inline interfaces/ifacetest/testiface.go]: I think you can move this just above the first method of this struct.

**Comment by bboozzoo** [comment]
Unit checks appear relevant:
```
# github.com/snapcore/snapd/interfaces/ifacetest_test
interfaces/ifacetest/testiface_test.go:213:8: spec.SetSlot undefined (type *hotplug.Specification has no field or method SetSlot)
interfaces/ifacetest/testiface_test.go:223:15: spec.Slot undefined (type *hotplug.Specification has no field or method Slot)

Crushing failure and despair.
```

**Comment by stolowski** [comment]
> Unit checks appear relevant:
> 
> ```
> # github.com/snapcore/snapd/interfaces/ifacetest_test
> interfaces/ifacetest/testiface_test.go:213:8: spec.SetSlot undefined (type *hotplug.Specification has no field or method SetSlot)
> interfaces/ifacetest/testiface_test.go:223:15: spec.Slot undefined (type *hotplug.Specification has no field or method Slot)
> 
> Crushing failure and despair.
> ```

Yep, expected, this PR depends on PR #5841 (which just landed)

File: interfaces/ifacetest/testiface.go (hunk 1 above)
Comment (stolowski) [reply interfaces/ifacetest/testiface.go]: Added a new comment above first method, left this one as this is consistent with TestInterface defined above.

**Pull request merged**